### PR TITLE
471 ui for adding consumables

### LINKED ISSUE #445: Better view for consumables
Would be good to display this a bit nicer...

image

_Originally posted by @jmbrunskill in https://github.com/msupply-foundation/unified-codes/issues/441#issuecomment-1853086084_

Might be best to revisit this once we've worked out what the input/categories of consumables will be!

### LINKED ISSUE #471: Add UI for adding new consumable/medical device

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #471 
Fixes #445 

## Description

Adds an input form for adding a new consumable. Similar structure to the drug input form, but with the different hierarchy.
Screenshot 2023-12-21 at 5 10 07 PM

Have made the 3 category inputs as dropdowns, but I don't know what the options available will be for that, so just used the one from Louisa's example lol. 
- Though now that I'm writing about it, I think I remember Louisa saying she wanted different intermediate and specific category options to be available depending on the higher level selection... might need to do something slightly different in the config?
  - Would be worth finding out how many different options there will be... perhaps if there's only a few, it's fine that they're not filtered by their parents?

Before we edit existing consumables with this form, we'll need Louisa to map those consumables to the new hierarchy types. 

Note: used the edit drug UI PR as a base... should review/merge that one first!

### CODE REVIEW

**Review by jmbrunskill — APPROVED** [review]
Code wise I think this is looking good, but I'm not convinced we have the heirachy correct yet.

For example if we look at the Pharmac spreadsheet as reference
image

In some cases it looks like the Category Level 1 should probably be our product?
For Haemodialysis they have a whole range of things though.

image

I'm not clear on what we're trying to achieve with the categories, but the product level should the level you're trying to search on in the UI I think.
It almost feels like categories should act more like Properties than nodes on the tree? I think this might have been Louisa' thought where she had the category branching off from the tree? Could still be looked up and referenced?
image


Would like to see a few more examples and the mappings for exisiting products before we take this too far...

**Comment by jmbrunskill** [comment]
@lache-melvin do you want to update this branch and remove the UNSPSC categories? I'm happy too if you want me to pic it up instead?

**Comment by jmbrunskill** [comment]
image

**Review by jmbrunskill — APPROVED** [review]
Looks good to me!
I'll tackle that duplication problem in issue #520 will tack on handling duplicates in the same request (maybe remove concurrency?)

File: frontend/system/src/Admin/EditEntity/ConsumableEditForm.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    // Convert the draft to a UpsertEntityInput type (stored within the change request until approved)
+    const entity = buildEntityFromConsumableInput(draft);
+
+    requestChange({
+      input: {
+        requestId: uuid(),
+        category: EntityCategory.Consumable,
+        changeType: initialEntity ? ChangeTypeNode.Change : ChangeTypeNode.New,
+        name: draft.name,
+        requestedFor: 'Country - coming soon', // TODO: capture this
```
[the remark this answers is not in the document]
Comment (lache-melvin) [reply frontend/system/src/Admin/EditEntity/ConsumableEditForm.tsx:62]: Yep! #517

**Pull request merged**